modules/storage-api/src/main/java/org/apache/ignite/internal/storage/engine/MvPartitionMeta.java

```diff
@@ -26,20 +27,50 @@
 public class MvPartitionMeta extends PrimitivePartitionMeta {
     private final byte[] groupConfig;
 
+    private final byte[] snapshotInfo;
+
     /** Constructor. */
     public MvPartitionMeta(
             long lastAppliedIndex,
             long lastAppliedTerm,
             byte[] groupConfig,
-            @Nullable LeaseInfo leaseInfo
+            @Nullable LeaseInfo leaseInfo,
+            byte[] snapshotInfo
     ) {
         super(lastAppliedIndex, lastAppliedTerm, leaseInfo);
 
         this.groupConfig = groupConfig;
+        this.snapshotInfo = snapshotInfo;
     }
 
     /** Returns replication group config as bytes. */
     public byte[] groupConfig() {
         return groupConfig;
     }
+
+    /** Returns snapshot info as bytes. */
+    public byte[] snapshotInfo() {
+        return snapshotInfo;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
```

Comment:
I think this won't work well because the parent class does the equality check for classes, not `instanceof`.

Reply:
Could you please clarify, why it won't work?

Reply:
There is a check for class equality here, but for the heir it will not work correctly.

```
    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        PrimitivePartitionMeta that = (PrimitivePartitionMeta) o;
        return lastAppliedIndex == that.lastAppliedIndex && lastAppliedTerm == that.lastAppliedTerm && Objects.equals(leaseInfo,
                that.leaseInfo);
    }
```

Reply:
Why won't it work correctly? Could you provide an example?